src/sentry/snuba/discover.py

```diff
@@ -770,82 +770,107 @@ def get_performance_facets(
-        orderby = []
+    sample_rate = options_sample_rate if sampling_enabled else None
+    sample_multiplier = sample_rate if sample_rate else 1
+
+    excluded_tags = [
+        "tags_key",
+        "NOT IN",
+        ["trace", "trace.ctx", "trace.span", "project", "browser"],
+    ]
+
+    with sentry_sdk.start_span(op="discover.discover", description="facets.aggregate_tags"):
+        conditions = snuba_filter.conditions
+        aggregate_comparison = transaction_aggregate * 1.01 if transaction_aggregate else 0
+        having = [excluded_tags]
+        if orderby and orderby in ("-sumdelta", "aggregate", "-aggregate"):
+            having.append(["aggregate", ">", aggregate_comparison])
+
+        if orderby is None:
+            orderby = []
+        else:
+            orderby = [orderby]
 
-    if aggregate_tags:
-        with sentry_sdk.start_span(op="discover.discover", description="facets.aggregate_tags"):
-            conditions = snuba_filter.conditions
-            conditions.append(["tags_key", "IN", aggregate_tags])
-            tag_values = raw_query(
-                aggregations=[
-                    [aggregate_function, aggregate_column, "aggregate"],
-                    ["count", None, "count"],
-                ],
-                conditions=conditions,
-                start=snuba_filter.start,
-                end=snuba_filter.end,
-                filter_keys=snuba_filter.filter_keys,
-                orderby=orderby + ["tags_key"],
-                groupby=["tags_key", "tags_value"],
-                dataset=Dataset.Discover,
-                referrer=referrer,
-                sample=sample_rate,
-                turbo=sample_rate is not None,
-                limitby=[TOP_VALUES_DEFAULT_LIMIT, "tags_key"],
-            )
-            results.extend(
+        tag_values = raw_query(
+            selected_columns=[
                 [
-                    PerformanceFacetResult(
-                        r["tags_key"], r["tags_value"], float(r["aggregate"]), int(r["count"])
-                    )
-                    for r in tag_values["data"]
-                ]
-            )
+                    "sum",
+                    [
+                        "minus",
+                        [
+                            aggregate_column,
+                            str(transaction_aggregate),
+                        ],
+                    ],
+                    "sumdelta",
+                ],
+            ],
+            aggregations=[
+                [aggregate_function, aggregate_column, "aggregate"],
+                ["count", None, "cnt"],
+            ],
+            conditions=conditions,
+            start=snuba_filter.start,
+            end=snuba_filter.end,
+            filter_keys=snuba_filter.filter_keys,
+            orderby=orderby + ["tags_key"],
+            groupby=["tags_key", "tags_value"],
+            having=having,
+            dataset=Dataset.Discover,
+            referrer=referrer,
```

Comment:
Can we use a different referrer for the two queries? We have this problem in the existing facets endpoint where we cant distinguish between the two already (something we need to fix).

Reply:
Oh good call 👍